cameras/pi-zero/large-display/software/camera/camera.py

```diff
@@ -104,18 +104,29 @@ def stop_video_recording(self):
     self.picam2.stop_recording()
 
   def change_mode(self, mode):
+    self.last_mode = mode
+
     if (mode == "zoom 1x"):
+      self.zoom_level = 1
+      self.pan_offset_x = 0
+      self.pan_offset_y = 0
       self.picam2.switch_mode(self.config_1x)
+      self.main.zoom_active = False
     elif (mode == "zoom 3x"):
+      self.zoom_level = 3
+      self.pan_offset_x = 1
+      self.pan_offset_y = 1
       self.picam2.switch_mode(self.config_3x)
     elif (mode == "zoom 7x"):
+      self.zoom_level = 7
+      self.pan_offset_x = 3
+      self.pan_offset_y = 3
       self.picam2.switch_mode(self.config_7x)
     elif (mode == "video"):
       self.picam2.switch_mode(self.video_config)
     else:
       self.picam2.switch_mode(self.config)
-    
-    self.last_mode = mode
+      self.last_mode = "zoom 1x" # reset

```

Comment:
I think this was important

Case of going full mode, last mode would be full... doesn't make sense

Should return to 1x zoom or previous zoom level